lib/collection/src/lib.rs

```diff
@@ -362,24 +365,41 @@ impl Collection {
             top: request.top,
         };
 
-        self.search(search_request, segment_searcher, search_runtime_handle)
-            .await
+        self.search(
+            search_request,
+            segment_searcher,
+            search_runtime_handle,
+            shard_selection,
+        )
+        .await
```

Comment:
this search request depends on vectors, retrieved from the whole collection, it won't work until distributed retrieve is not implemented.

Reply:
I don't understand this.
Following the call stack, the search call ends up on the `LocalShard` if the `shard_selection` is valid.
Could you please elaborate?

Reply:
Received the explanation offline :memo: 

`recommend` calls `retrieve` first to fetch the vectors, so the `shard_selection` must be propagated to `retrieve`.

https://github.com/qdrant/qdrant/pull/463/commits/c70afa33f81c5bb94bfc95ee19b159657848f11f

Reply:
https://github.com/qdrant/qdrant/issues/473